query-engine/connectors/query-connector/src/interface.rs

```diff
@@ -295,9 +310,10 @@ pub trait WriteOperations {
         trace_id: Option<String>,
     ) -> crate::Result<usize>;
 
-    /// Delete single record in the `Model` with the given `Filter`.
-    /// Return selected fields of the deleted record, if the connector
-    /// supports it. If the connector does not support it, error is returned.
+    /// Delete single record in the `Model` with the given `Filter` and returns
+    /// selected fields of the deleted record.
+    /// This method should not be used if the connector does not support returning
```

Comment:
The reason for a change in wording is that we cannot check compatibilities on the SQL connector layer. So if somebody accidentally calls this method on an unsupported connector like MSSQL, we will proceed and try to generate a query.